node/pkg/altpub/alternate_pub_test.go

```diff
@@ -0,0 +1,279 @@
+		output  string
+	}
+	tests := []test{
+		{inputUs: 0, output: "immediate"},
+		{inputUs: 250, output: "250µs"},
+		{inputUs: 42000, output: "42ms"},
+		{inputUs: 7000000, output: "7s"},
+	}
+
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("%d", tc.inputUs), func(t *testing.T) {
+			delay := time.Microsecond * time.Duration(tc.inputUs)
+			require.Equal(t, tc.output, Delay(delay).String())
+		})
+	}
+}
+
+func TestEnabledChainsString(t *testing.T) {
+	// NOTE: The output will be sorted in chainID order.
+	type test struct {
+		input     []vaa.ChainID
+		exceptFor bool
+		output    string
+	}
+	tests := []test{
+		{input: []vaa.ChainID{}, output: "all-chains"},
+		{input: []vaa.ChainID{vaa.ChainIDPythNet}, output: "pythnet"},
+		{input: []vaa.ChainID{vaa.ChainIDEthereum, vaa.ChainIDSolana}, output: "solana,ethereum"},
+		{input: []vaa.ChainID{vaa.ChainIDPythNet}, exceptFor: true, output: "all-except:pythnet"},
+		{input: []vaa.ChainID{vaa.ChainIDEthereum, vaa.ChainIDSolana}, exceptFor: true, output: "all-except:solana,ethereum"},
+	}
+
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("%d", tc.input), func(t *testing.T) {
+			enabledChainsMap := make(map[vaa.ChainID]struct{})
+			for _, chainID := range tc.input {
+				enabledChainsMap[chainID] = struct{}{}
+			}
+			require.Equal(t, tc.output, EnabledChains{tc.exceptFor, enabledChainsMap}.String())
+		})
+	}
+}
+
+func TestParseEndpoint(t *testing.T) {
+	type test struct {
+		input     string
+		errText   string
+		delayUs   int64
+		chains    []vaa.ChainID
+		exceptFor bool
+		label     string
+	}
+	tests := []test{
+		// Success cases
+		{label: "Minimum_config", input: "test;http://localhost:3333"},
+		{label: "With_delay_no_chains", input: "test;http://localhost:3333;1s", delayUs: 1000000},
+		{label: "With_delay_and_one_chain_number", input: "test;http://localhost:3333;200us;1", delayUs: 200, chains: []vaa.ChainID{vaa.ChainIDSolana}},
+		{label: "With_delay_and_one_chain_name", input: "test;http://localhost:3333;200us;solana", delayUs: 200, chains: []vaa.ChainID{vaa.ChainIDSolana}},
+		{label: "With_delay_and_two_chain_numbers", input: "test;http://localhost:3333;500ms;1,2", delayUs: 500000, chains: []vaa.ChainID{vaa.ChainIDSolana, vaa.ChainIDEthereum}},
+		{label: "With_delay_and_chain_number_and_name", input: "test;http://localhost:3333;500ms;1,ethereum", delayUs: 500000, chains: []vaa.ChainID{vaa.ChainIDSolana, vaa.ChainIDEthereum}},
+		{label: "With_no_delay_and_one_chain", input: "test;http://localhost:3333;0;1", chains: []vaa.ChainID{vaa.ChainIDSolana}},
+		{label: "With_no_delay_and_except_one_chain", input: "test;http://localhost:3333;0;-pythnet", chains: []vaa.ChainID{vaa.ChainIDPythNet}, exceptFor: true},
+		{label: "With_no_delay_and_except_two_chains", input: "test;http://localhost:3333;0;-pythnet,1", chains: []vaa.ChainID{vaa.ChainIDSolana, vaa.ChainIDPythNet}, exceptFor: true},
+
+		// Error cases
+		{label: "Empty", input: "", errText: "not enough fields"},
```

Comment:
It could be good to add a length == 3 case here, e.g. input "wormholescan;http://localhost:3333;0". This is in reference to the README where specifying a delay means we need to specify a chain.